### PR TITLE
[DOC] Update ingest pipelines content and page navigation

### LINKED ISSUE #4193: [DOC] Add ingest processors
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Add ingest processors; new content, several processors

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Updates ingest pipelines content and page navigation

### Issues Resolved
#4193
#4283

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 1]
```diff
@@ -36,48 +34,54 @@ POST /_ingest/pipeline/35678/_simulate
 ```
 {% include copy-curl.html %}
 
-## Path and HTTP methods
+## Example requests
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:37]: This is not just a complete example, but PATH and HTTP method. You need a body for the request.

```
GET _ingest/pipeline/_simulate
POST _ingest/pipeline/_simulate
```

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 2]
```diff
@@ -36,48 +34,54 @@ POST /_ingest/pipeline/35678/_simulate
...
 POST _ingest/pipeline/_simulate
 ```
 
-Simulate a single pipeline based on the pipeline's ID.
+Requests to simulate a single pipeline based on the pipeline identifier.
 
 ```
-GET _ingest/pipeline/{id}/_simulate
-POST _ingest/pipeline/{id}/_simulate
+GET _ingest/pipeline/pipeline-id/_simulate
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:49]: Same here. This is not an example. Missing request body.

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 3]
```diff
@@ -36,48 +34,54 @@ POST /_ingest/pipeline/35678/_simulate
...
+POST _ingest/pipeline/pipeline-id/_simulate
 ```
 
-## URL parameters
+## Path parameters
+
+Parameter | Type | Description
+:--- | :--- | :---
+`pipeline` | String | Pipeline identifier or wildcard expression of pipeline identifiers used to limit the request.
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:57]: Wildcard does not work here. I feel like the previous format is better than having a separate `path parameters` section.

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 4]
```diff
@@ -36,48 +34,54 @@ POST /_ingest/pipeline/35678/_simulate
 ```
 {% include copy-curl.html %}
 
-## Path and HTTP methods
+## Example requests
 
-Simulate the last ingest pipeline created.
+Requests to simulate the lastest ingest pipeline created:
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:39]:
```suggestion
Requests to simulate the latest ingest pipeline created:
```

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 5]
```diff
@@ -36,48 +34,54 @@ POST /_ingest/pipeline/35678/_simulate
...
 
 ## Request body fields
 
-Field | Required | Type | Description
-:--- | :--- | :--- | :---
-`pipeline` | Optional | object | The pipeline you want to simulate. When included without the pipeline `{id}` inside the request path, the response simulates the last pipeline created.
-`docs` | Required | array of objects | The documents you want to use to test the pipeline.
+Field | Type | Description
+:--- | :--- | :---
+`pipeline` | Object | The pipeline to be simulated. If the pipeline identifier is not included, then the response simulates the lastest pipeline created. Optional.
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:71]:
```suggestion
`pipeline` | Object | The pipeline to be simulated. If the pipeline identifier is not included, then the response simulates the latest pipeline created. Optional.
```

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 6]
```diff
@@ -36,48 +34,54 @@ POST /_ingest/pipeline/35678/_simulate
...
 The `docs` field can include the following subfields:
 
-Field | Required | Type | Description
+Field | Type | Description
 :--- | :--- | :---
-`id` | Optional |string | An optional identifier for the document. The identifier cannot be used elsewhere in the index.
-`index` | Optional | string | The index where the document's transformed data appears.
-`source` | Required | object | The document's JSON body.
+`id` | String | A unique identifier for a document. The identifier cannot be used elsewhere in the index. Optional.
+`index` | String | A collection of documents that have undergone specfic data transformation within an ingest pipeline. Optional.
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:79]: Not all of the documents in the index might have undergone the data transformation. The index is only to tell where this document will get ingested.

```suggestion
`index` | String | The index where the document's transformed data appears. Optional.
```

File: _api-reference/ingest-apis/simulate-ingest.md [hunk 7]
```diff
@@ -149,9 +153,9 @@ Responses vary based on which path and HTTP method you choose.
 }
 ```
 
-### Receive verbose response 
+### Receive a verbose response 
 
-With the `verbose` parameter set to `true`, the response shows how each processor transforms the specified document. 
+When the `verbose` parameter is set to `true`, the response shows the prcoessor results, that is, how the processor transformed the data.
```
Comment (heemin32) [inline _api-reference/ingest-apis/simulate-ingest.md:158]:
```suggestion
When the `verbose` parameter is set to `true`, the response shows the processor results, that is, how the processor transformed the data. 
```

File: _api-reference/ingest-apis/create-update-ingest.md [hunk 8]
```diff
@@ -29,51 +29,30 @@ PUT _ingest/pipeline/12345
 ```
 {% include copy-curl.html %}
 
-## Path and HTTP methods
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _api-reference/ingest-apis/create-update-ingest.md:32]: Added in.

File: _api-reference/ingest-apis/create-update-ingest.md [hunk 9]
```diff
@@ -29,51 +29,30 @@ PUT _ingest/pipeline/12345
...
-
-## Response
-
-```json
-{
-  "acknowledged" : true
-}
-```
-
+`pipeline` | String | Pipeline ID or wildcard expression of pipeline IDs used to limit the request.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _api-reference/ingest-apis/create-update-ingest.md:52]: Revised. Please verify accuracy.

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+          "value" : "value"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Delete a pipeline
+
```
Comment (vagimeli) [inline _api-reference/ingest-apis/ingest-pipelines.md:280]: Processor documentation in progress. See [#4193](https://github.com/opensearch-project/documentation-website/issues/4193)

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+  "pipeline-id" : {
+    "description" : "Optional description of the pipeline",
+    "processors" : [
+      {
+        "set" : {
+          "field" : "field-name",
+          "value" : "value"
+        }
+      }
+    ]
```
Comment (vagimeli) [inline _api-reference/ingest-apis/ingest-pipelines.md:274]: @heemin32 Do we want to include a section about creating a pipeline on an existing index?

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+    {
+      "lowercase": {
+        "field": "my-keyword-field"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
```
Comment (vagimeli) [inline _api-reference/ingest-apis/ingest-pipelines.md:59]: @heemin32 Do we want to include content about if a pipeline fails? Are there common reasons why a pipeline fails?

File: _api-reference/ingest-apis/index.md [hunk 13]
```diff
@@ -9,6 +9,160 @@ redirect_from:
 
 # Ingest APIs
```
Comment (vagimeli) [inline _api-reference/ingest-apis/index.md:11]: @kolchfa-aws Please provide a peer review of this PR, if possible by end of week. Reach out if we need to sync on it. Thank you!

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+`description` | Optional | String | Description of the ingest pipeline. 
+`processors` | Required | Array | The processor that performs a transformation on the documents. Processors run sequentially. 
+
+### Parameters
+
+The following table lists the parameters used with creating, or updating, a pipeline.
+
+Parameter | Required | Type | Description
+:--- | :--- | :--- | :---
+`pipeline` | Required | String | The unique identifier, or pipeline id, assigned to the ingest pipeline. A pipeline id is used in API requests to specify which pipeline should be created or modified.
```
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:44]:
```suggestion
`pipeline-id` | Required | String | The unique identifier, or pipeline ID, assigned to the ingest pipeline. A pipeline id is used in API requests to specify which pipeline should be created or modified.  
```

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+## Delete a pipeline
+
+Use the following requests to delete pipelines. 
+
+#### Example HTTP methods
+
+**Delete a specific pipeline**
+
+```json
+DELETE /_ingest/pipeline/pipeline-id
```
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:290]:
```suggestion
DELETE /_ingest/pipeline/<pipeline-id>
```

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 16]
```diff
@@ -0,0 +1,309 @@
...
+    ]
+  }
+}
+```
+
+## Delete a pipeline
+
+Use the following requests to delete pipelines. 
+
+#### Example HTTP methods
```
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:285]: Remove this heading

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 17]
```diff
@@ -0,0 +1,309 @@
...
+
+**Delete a specific pipeline**
```
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:287]:
```suggestion
To delete a specific pipeline, pass the pipeline ID as a parameter:
```

File: _api-reference/ingest-apis/ingest-pipelines.md (hunk 15 above)
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:290]:
```suggestion
DELETE /_ingest/pipeline/<pipeline-id>
```

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 18]
```diff
@@ -0,0 +1,309 @@
...
+
+```json
+DELETE /_ingest/pipeline/pipeline-id
+```
+{% include copy-curl.html %}
+
+**Delete all pipelines**
```
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:294]:
```suggestion
To delete all pipelines in the cluster, use the wildcard character (`*`):
```

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 19]
```diff
@@ -0,0 +1,309 @@
...
+
+```json
+DELETE /_ingest/pipeline/*
+```
+{% include copy-curl.html %}
+
+To delete all pipelines in a cluster, the asterisk (*) is necessary.
```
Comment (kolchfa-aws) [inline _api-reference/ingest-apis/ingest-pipelines.md:301]:
```suggestion
```

**Review by kolchfa-aws — COMMENTED** [review]
Please see my comments and let me know if you'd like to discuss. I would also create more explicit examples for this documentation. I'll help create examples offline.

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 20]
```diff
@@ -0,0 +1,382 @@
...
+
+```json
+DELETE /_ingest/pipeline/*
+```
+{% include copy-curl.html %}
+
+
+## Create a pipeline on an existing index
+
+_<Do we want to include a section on creating a pipeline on an existing index?>
```
Comment (vagimeli) [inline _api-reference/ingest-apis/ingest-pipelines.md:378]: @heemin32 Do we want to include a section about creating a pipeline on an existing index?

Comment (heemin32) [reply _api-reference/ingest-apis/ingest-pipelines.md:385]: As long as I know, there is no such thing like creating a pipeline on an existing index.
Once we create a ingest pipeline, we can use the pipeline to add a document to an existing index.

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 21]
```diff
@@ -0,0 +1,390 @@
...
+
+# Ingest pipelines
+
+Ingest pipelines in OpenSearch are managed by using ingest API operations. When using ingest in production environments, your cluster should contain at least one node with the node roles permission set to `ingest`. For more information about setting up node roles within a cluster, see [Cluster formation]({{site.url}}{{site.baseurl}}/opensearch/cluster/).
+
+## Prerequisites
+
+The following are prerequisites for using OpenSearch ingest pipelines:
+
+- <SME insert information>
```
Comment (vagimeli) [inline _api-reference/ingest-apis/ingest-pipelines.md:22]: @heemin32 Please provide any prerquisites details we should include in the documentation. We're aiming to publish this documentation this week. If we need to discuss offline, please let me know. Thank you.

File: _api-reference/ingest-apis/ingest-pipelines.md [hunk 22]
```diff
@@ -0,0 +1,390 @@
...
+  - /opensearch/rest-api/ingest-apis/create-update-ingest/
+  - /api-reference/ingest-apis/create-update-ingest/
+  - /api-reference/ingest-apis/get-ingest/
+  - /api-reference/ingest-apis/simulate-ingest/
+  - /api-reference/ingest-apis/delete-ingest/
+---
+
+# Ingest pipelines
+
+Ingest pipelines in OpenSearch are managed by using ingest API operations. When using ingest in production environments, your cluster should contain at least one node with the node roles permission set to `ingest`. For more information about setting up node roles within a cluster, see [Cluster formation]({{site.url}}{{site.baseurl}}/opensearch/cluster/).
```
Comment (heemin32) [inline _api-reference/ingest-apis/ingest-pipelines.md:16]: `ingest` is a role of a node but not permission.

```suggestion
Ingest pipelines in OpenSearch are managed by using ingest API operations. When using ingest in production environments, your cluster should contain at least one node with the `ingest` role. For more information about setting up node roles within a cluster, see [Cluster formation]({{site.url}}{{site.baseurl}}/opensearch/cluster/).
```

File: _api-reference/ingest-apis/ingest-pipelines.md (hunk 21 above)
Comment (heemin32) [inline _api-reference/ingest-apis/ingest-pipelines.md:22]: I think this line from description above could be one of prerequisites. `When using ingest in production environments, your cluster should contain at least one node with the `ingest` role.`

In addition to that, following prerequisites is needed when security features are enabled.
```
If the OpenSearch security features are enabled, you must have the `cluster_manage_pipelines` permission to manage ingest pipelines.
```

**Comment by vagimeli** [comment]
See PR #4299. Revised pipeline documentation included in that PR.

**Review by heemin32 — COMMENTED** [review]
Will continue the review in another PR.

File: _api-reference/ingest-apis/ingest-pipelines.md (hunk 22 above)
Comment (vagimeli) [reply _api-reference/ingest-apis/ingest-pipelines.md:16]: Changed in PR #4299

File: _api-reference/ingest-apis/ingest-pipelines.md (hunk 21 above)
Comment (vagimeli) [reply _api-reference/ingest-apis/ingest-pipelines.md:22]: Revised in PR #4299

**Pull request closed without merging**